### ISSUE TITLE
RFE: Check filesize of files referenced by companion.ome.xml

### ISSUE THREAD

**Issue by dominikl** [ome/bioformats#4010]
If a companion file references an empty file (0 bytes) it will fail on import, but not on `import -f` or by `showinf -nopix`.
I wonder if it's possible that Bioformats will pick up on this when it gathers the files which have to be imported?
I noticed that when creating companion.ome.xml files for idr0119 based on a file listing. One of them for example
referes to 
```
<TiffData FirstC="1" FirstT="23" FirstZ="0" PlaneCount="1">
        <UUID FileName="HC01001/C2/field_2/HC01001_G_C2_2_00d11h30m.tif">urn:uuid:f5bcb507-5893-44b8-94ed-81bded22a892</UUID>
</TiffData>
```
The file exists, but it's empty:
```
-rw-rw----. 1 6166 idrnfs     0 Nov 22 03:30 HC01001_G_C2_2_00d11h30m.tif
```
But wether `showinf` nor `import -f` fails on it, only the real import (which in that case nearly takes a full day).

**Comment by melissalinkert** [comment]
This can be reproduced by taking the public plate companion dataset (https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/plate-companion/) and replacing `well-C2-2.ome.tiff` with a 0-byte file of the same name.

The first `showinf hcs.companion.ome` will show that `well-C2-2.ome.tiff` is not included in the used files list; there is a logged warning that this file is invalid. The used file list reflects what would be transferred during OMERO import. Removing the empty `well-C2-2.ome.tiff` file from the directory and repeating `showinf -debug hcs.companion.ome` throws:

```
loci.formats.FormatException: Missing file well-C2-2.ome.tiff associated with UUID 
          urn:uuid:6dd96c31-0f09-402a-918f-d691a7ef43ca
        .
	at loci.formats.in.OMETiffReader.initFile(OMETiffReader.java:832)
	at loci.formats.FormatReader.setId(FormatReader.java:1466)
	at loci.formats.ImageReader.setId(ImageReader.java:863)
	at loci.formats.ReaderWrapper.setId(ReaderWrapper.java:660)
	at loci.formats.Memoizer.setId(Memoizer.java:726)
	at loci.formats.ReaderWrapper.setId(ReaderWrapper.java:660)
	at loci.formats.tools.ImageInfo.testRead(ImageInfo.java:1043)
	at loci.formats.tools.ImageInfo.main(ImageInfo.java:1129)
```

#4073 is one approach to solving this, which will fail on initial `setId` if any invalid OME-TIFF is encountered. That is currently excluded from the build, as I suspect we want to discuss whether it makes sense to fail immediately by default or successfully initialize and include the empty file on the used file list so that as much of the data as possible can be read.

**Comment by melissalinkert** [comment]
Closing as #4073 was merged for 7.0.1, so import and `showinf` should now fail quickly in this case.
